ARMmbed/theia
# Dirty saveall

#### What it does


This PR adds a check to only save open files if they are dirty (as the function describes it should). Without this, all open files were forced to be saved, causing filewatchers to incorrectly fire updates.

#### How to test


Check modified dates on files which are open when a debug session is launched. They should only update if the file was changed.

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
